common/flatpak-dir.c

```diff
@@ -2388,6 +2388,35 @@ flatpak_deploy_data_get_appdata_license (GVariant *deploy_data)
   return flatpak_deploy_data_get_string (deploy_data, "appdata-license");
 }
 
+const char *
+flatpak_deploy_data_get_appdata_content_rating_type (GVariant *deploy_data)
+{
+  g_autoptr(GVariant) metadata = g_variant_get_child_value (deploy_data, 4);
+  const char *value = NULL;
+
+  g_variant_lookup (metadata, "appdata-content-rating", "(&s*)", &value, NULL);
+
+  return value;
+}
+
+GHashTable *  /* (transfer container) */
+flatpak_deploy_data_get_appdata_content_rating (GVariant *deploy_data)
+{
+  g_autoptr(GVariant) metadata = g_variant_get_child_value (deploy_data, 4);
+  const char *id = NULL, *val = NULL;
+  g_autoptr(GHashTable) content_rating = g_hash_table_new_full (g_str_hash, g_str_equal, NULL, NULL);
+  g_autoptr(GVariantIter) iter = NULL;
+
+  g_variant_lookup (metadata, "appdata-content-rating", "(*a{ss})", NULL, &iter);
```

Comment:
I think you either need to handle this returning FALSE, or the iter being NULL below, in case the key is not in the metadata.

Reply:
Good catch. Fixed locally.